### ISSUE TITLE
Haddocks don't build for classy-prelude-yesod

### ISSUE THREAD

**Issue by MaxGabriel** [snoyberg/mono-traversable#169]
I'm pretty sure this is just a Haddock bug and you can close this, but just for awareness sake:

```
stack haddock classy-prelude-yesod --no-haddock-deps # no-haddock-deps to avoid haskell-src-exts using up all memory https://github.com/haskell/haddock/issues/928
```

```
Haddock coverage:
haddock: internal error: internal: extractDecl
CallStack (from HasCallStack):
  error, called at utils/haddock/haddock-api/src/Haddock/Interface/Create.hs:1097:12 in main:Haddock.Interface.Create
Completed 8 action(s).

--  While building custom Setup.hs for package classy-prelude-yesod-1.4.0 using:
      /Users/maximiliantagher/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_2.2.0.1_ghc-8.4.3 --builddir=.stack-work/dist/x86_64-osx/Cabal-2.2.0.1 haddock --html --hoogle --html-location=../$pkg-$version/ --haddock-option=--hyperlinked-source
    Process exited with code: ExitFailure 1
```

LTS: 12.4
OS: macOS Mojave

**Comment by snoyberg** [comment]
I agree, it's likely a Haddock bug. Let's leave it open in case (1) someone wants to implement a workaround and (2) so future reporters can find this issue.

**Comment by normenmueller** [comment]
Is there anything new on this? Haddock still fails on `classy-prelude-yesod`. Any workaround? Otherwise my `stack hoogle` always fails.

**Comment by snoyberg** [comment]
I've made no changes and don't intend to. I've generally given up on Haddock to be honest.

**Comment by normenmueller** [comment]
@snoyberg off topic: I'm also not bothered w/ Haddock too much, but `stack hoogle` quits in case of Haddock errors. Is there any workaround, as I'm bothered about Hoogle ;-)

**Comment by snoyberg** [comment]
Not that I'm aware of

**Comment by normenmueller** [comment]
@snoyberg Thanks for your time anyway. Re-posted at https://stackoverflow.com/questions/55479788/build-local-hoogle-db-despite-error-in-haddock

**Comment by MaxGabriel** [comment]
It might be this bug https://github.com/haskell/haddock/issues/979

**Comment by normenmueller** [comment]
Thanks for the link. Contributing to this thread now.

**Comment by meteficha** [comment]
> I've made no changes and don't intend to. I've generally given up on Haddock to be honest.

What have you been using instead of Haddock? These bugs are annoying but not having anything is worse :).

**Comment by snoyberg** [comment]
I still use Haddock when it's available. I've just spent so much time working around it that, if I don't have the Haddocks, I'll drop down to reading the source directly. It sucks, I agree.
